src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -38,9 +38,16 @@ using namespace htm;
 
 
 // work-load
-Real64 BenchmarkHotgym::run(UInt EPOCHS, bool useSPlocal, bool useSPglobal, bool useTM, const UInt COLS, const UInt DIM_INPUT, const UInt CELLS) {
+Real64 BenchmarkHotgym::run(UInt EPOCHS, bool useSPlocal, bool useSPglobal, bool useTM, const UInt COLS, const UInt DIM_INPUT, const UInt CELLS)
+{
 #ifndef NDEBUG
-  EPOCHS = 2; // make test faster in Debug
+EPOCHS = 2; // make test faster in Debug
+#endif
+
+#if defined __aarch64__ || defined __arm__
+#undef _ARCH_DETERMINISTIC
```

Comment:
interesting that it fails on `aarch64`. Ok to do it as is here, but I wonder why...? We took special care and wrote custom Random for this to be platform-agnostic and deterministic. Other algos should be determ according to the c++ standard.

Reply:
Perhaps the multiarch/qemu nature of our solution adds some noise?  That's probably a lot of software emulation instead of actual hardware.  Anyway, sorry, not sure :(